docs/03-architecture/05-serialization.md

```diff
@@ -5,42 +5,24 @@ sidebar_position: 5
-framework. We provide first-class support for json
-through [`cosmwasm::serde`](https://github.com/CosmWasm/cosmwasm/blob/master/src/serde.rs)
-and [`cosmwasm-template`](https://github.com/CosmWasm/cosmwasm-template), but anyone can swap this out - provided they
-provide client support for the format.
+The natural solution was to use JSON everywhere. It is self-describing, human-readable, and used in APIs everywhere. It does have some downsides, such as handling numbers over 2^53 (just use strings), no clear distinction between strings and base64-encoded binary, and no hard-coded schema. We auto-generate [JSON Schema](https://json-schema.org/) descriptors for the [public API of contracts](https://github.com/CosmWasm/cosmwasm-examples/tree/master/escrow/schema), which can be used to inspect the supported API and optionally used in client side tooling for auto-validation of messages.
+
+The feedback when developing and debugging with this has been positive, and we are quite happy with the Developer UX with this. It is too early to tell if the message size and free-form schema will become a hinderance in production. However, please note that contracts define their own parsing logic for messages, the codec is not enforced by the framework. We provide first-class support for json through [`cosmwasm::serde`](https://github.com/CosmWasm/cosmwasm/blob/master/src/serde.rs) and [`cosmwasm-template`](https://github.com/CosmWasm/cosmwasm-template), but anyone can swap this out - provided they provide client support for the format.
 
 It is helpful to have consistency to aid client development, as well as contract-contract calls.
 
 ## Protobuf {#protobuf}
 
-Protobuf is a well-known and widely-supported binary format. It gives a stricter schema guarantees than JSON and more
-compact format. Protocol Buffers and GRPC support has been added with Cosmos SDK v0.39.0 upgrade.
+Protobuf is a well-known and widely-supported binary format. It gives a stricter schema guarantees than JSON and more compact format. Protocol Buffers and GRPC support has been added with Cosmos SDK v0.39.0 upgrade.
 
 ## Cap'n Proto {#capn-proto}
 
-[Cap'n Proto](https://capnproto.org/) is a super-lean encoding format with zero-copy reads, and no parsing needed. This
-has been [suggested for use in CosmWasm](https://github.com/CosmWasm/cosmwasm/issues/78) as an optional addition. This
-may be considered as an opt-in format for contracts desiring such efficiency or strict schema, or possibly just used for
-encoding internal data structures (`Params`).
+[Cap'n Proto](https://capnproto.org/) is a super-lean encoding format with zero-copy reads, and no parsing needed. This has been [suggested for use in CosmWasm](https://github.com/CosmWasm/cosmwasm/issues/78) as an optional addition. This may be considered as an opt-in format for contracts desiring such efficiency or strict schema, or possibly just used for encoding internal data structures (`Params`).
```

Comment:
What command did you use to reformat all the MD files?

It would be good to add that somewhere (CONTRIBUTING.md?) so all controls do the same guidelines and we have smaller diffs.

Reply:
Ah, I write a lot of prose, so just a keybinding in editor to put cursors at the ends of all lines in a region. I gather that the preferred style here is actually to wrap to 120chrs, however